ci/docker.sh

```diff
@@ -29,46 +32,95 @@ fi
-  if [ "${prod_release}" = "true" ]; then
+  # Determine image tags based on release type
+  if [[ "${prod_release}" == "true" ]]; then
     docker_tag_node="$(cut -d '-' -f1 <<< "${docker_tag_full}")"
     publish_tags=("${docker_tag_full}" "${docker_tag_node}" "latest")
-  elif [ "${dev_release}" = "true" ]; then
+  elif [[ "${dev_release}" == "true" ]]; then
     docker_tag_node="$(cut -d '-' -f1 <<< "${docker_tag_full}")-$(cut -d '-' -f3- <<< "${docker_tag_full}")"
     publish_tags=("${docker_tag_full}" "${docker_tag_node}")
-  elif [ "${test_release}" = "true" ]; then
+  elif [[ "${test_release}" == "true" ]]; then
     publish_tags=("${docker_tag_full}")
+  else
+    fn_die "ERROR: No valid release type specified for IS_A_RELEASE=true."
   fi
 
+  # Build and publish images to DockerHub (and private repo if set)
+  # Build the Docker image
+  log_info "=== Building Docker image: ${docker_image_build_name}:${docker_tag_full} ==="
+  docker build --build-arg PROFILE=production -f "${docker_file_path}" -t "${docker_image_build_name}:${docker_tag_full}" .
+
+  # Login to DockerHub
+  log_info "=== Publishing Docker image(s) on Docker Hub ==="
+  echo "${docker_hub_token}" | docker login -u "${docker_hub_username}" --password-stdin
+
+  # Tag and push to DockerHub
   for publish_tag in "${publish_tags[@]}"; do
-    log_info "Publishing docker image: ${docker_image_build_name}:${publish_tag}"
-    docker tag "${docker_hub_org}/${docker_image_build_name}:${docker_tag_full}" "index.docker.io/${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
-    docker push "index.docker.io/${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
+    log_info "Publishing docker image to Docker Hub: ${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
+    docker tag "${docker_image_build_name}:${docker_tag_full}" "${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
+    docker push "${docker_hub_org}/${docker_image_build_name}:${publish_tag}"
   done
+
+  # Publish to Private Repository if specified
+  if [[ -n "${private_docker_repo:-}" ]]; then
+    log_info "=== Publishing Docker image(s) to Private Repository: ${private_docker_repo} ==="
+    if [[ -z "${DOCKER_HUB_TOKEN:-}" || -z "${DOCKER_HUB_USERNAME:-}" ]]; then
+      fn_die "ERROR: DOCKER_HUB_USERNAME and DOCKER_HUB_TOKEN must be set for private repository operations."
+    fi
+    echo "${DOCKER_HUB_TOKEN}" | docker login -u "${DOCKER_HUB_USERNAME}" --password-stdin "${private_docker_repo}"
+
+    for publish_tag in "${publish_tags[@]}"; do
+      log_info "Publishing docker image to Private Repository: ${private_docker_repo}:${publish_tag}"
+      docker tag "${docker_image_build_name}:${docker_tag_full}" "${private_docker_repo}:${publish_tag}"
+      docker push "${private_docker_repo}:${publish_tag}"
+    done
+  fi
+
+elif [[ "${is_a_release}" == "false" && "${dev_release}" == "true" && -n "${private_docker_repo:-}" ]]; then
```

Comment:
I think we should not be building anything if `is_a_release == false` The idea of `is_a_relese = true` was that if IT IS TRUE then it must be one of the `dev,prod,test` OTHERWISE it is FALSE and we exit

Reply:
You are right, but in this case we are also allowing the building for main merges to be able to execute e2e testing
cc: @yurikoinaba

Reply:
We can discuss this in details if needed, so I can probably understand the intention better :pray: